tests/integration/test_conftest/test_conftest.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+"""
+The aim of this module is to test db isolation between single tests.
+The `app` fixture does not provide db isolation between single tests: this
+means that if a test adds an object to the db, the same obj is found
+in the next tests.
+The `isolated_app` fixture instead provides db isolation between single tests.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+from flask_sqlalchemy import models_committed
+
+from invenio_db import db
+from invenio_oauthclient.models import User
+from invenio_pidstore.models import PersistentIdentifier
+
+
+@pytest.fixture(scope='module')
+def pids_count(_app):
+    # Count the # PIDs currently loaded in the demosite.
+    return PersistentIdentifier.query.count()
+
+
+@pytest.fixture(scope='module')
+def users_count(_app):
+    # Count the # User currently loaded in the demosite.
+    return User.query.count()
+
+
+# Test `app` fixture.
+
+
+def test_app_fixture_lacks_db_isolation_step1(pids_count, app):
+    assert PersistentIdentifier.query.count() == pids_count
+
+    PersistentIdentifier.create(
+        pid_type='type1',
+        pid_value='value1',
+    )
+    # The #PIDs must have incremented.
+    assert PersistentIdentifier.query.count() == pids_count + 1
+
+
+def test_app_fixture_lacks_db_isolation_step2(pids_count, app):
+    assert PersistentIdentifier.query.count() == pids_count + 1
+    # Force the cleanup.
+    PersistentIdentifier.get(
+        pid_type='type1',
+        pid_value='value1',
+    ).delete()
```

Comment:
why not do the cleanup in the previous test after the `assert`? having inter-test dependencies is probably a bad idea.

Reply:
The point of this test is to prove that the `app` fixture lacks db isolation (as the test name suggests). Because of this lack the `app` fixture might introduce inter-test dependencies which indeed a bad idea. The goal of this entire code change is to provide an `isolated_app` which .........

Reply:
:+1: I must have misread it yesterday